app/app_test.go

```diff
@@ -12,39 +13,37 @@ func TestInitNavigationKeys(t *testing.T) {
 	a := localMock.NewMockApplication(p)
 
 	a.On("PageNavigator").Return(p)
-	a.On("Exit").Return()
-	a.On("Forward").Return()
-	a.On("Back").Return()
-	a.On("NextChapter").Return()
-	a.On("PrevChapter").Return()
-
-	p.On("ScrollDown").Return()
-	p.On("ScrollUp").Return()
-	p.On("ScrollLeft").Return()
-	p.On("ScrollRight").Return()
-	p.On("ToTop").Return()
-	p.On("ToBottom").Return()
 
 	keymap, chmap := initNavigationKeys(a)
 
-	keymap[termbox.KeyArrowDown]()
-	keymap[termbox.KeyArrowUp]()
-	keymap[termbox.KeyArrowRight]()
-	keymap[termbox.KeyArrowLeft]()
-	keymap[termbox.KeyEsc]()
-
-	chmap['j']()
-	chmap['k']()
-	chmap['h']()
-	chmap['l']()
-	chmap['g']()
-	chmap['G']()
-
-	chmap['q']()
-	chmap['f']()
-	chmap['b']()
-	chmap['L']()
-	chmap['H']()
-
-	p.AssertExpectations(t)
+	var zeroValueTermboxKey termbox.Key
+
+	verifyMethodCall := func(receiver string, methodName string, letterKey rune, tbKey termbox.Key) {
+		if receiver == "pager" {
+			p.On(methodName).Return()
+		}
+		if receiver == "app" {
+			a.On(methodName).Return()
+		}
+
+		if unicode.IsLetter(letterKey) {
```

Comment:
What's the thinking behind this check? If it's not a letter should the test fail?

Reply:
Yeah, it's not ideal. Since for now we are using letters for navigation... but I've got a better idea comming in a following commit.

Reply:
Fixed by #13.